junit-platform-launcher/src/main/java/org/junit/platform/launcher/Launcher.java

```diff
@@ -81,8 +81,24 @@
 	 * collecting their results, and notify {@linkplain #registerTestExecutionListeners
 	 * registered listeners} about the progress and results of the execution.
 	 *
+	 * <p>The default implementation delegates to
+	 * {@link #execute(LauncherDiscoveryRequest, TestExecutionListener...)} with no
+	 * additional listeners.
+	 *
 	 * @param launcherDiscoveryRequest the launcher discovery request; never {@code null}
 	 */
-	void execute(LauncherDiscoveryRequest launcherDiscoveryRequest);
+	default void execute(LauncherDiscoveryRequest launcherDiscoveryRequest) {
```

Comment:
Deprecate it in M4 and remove it in M5 (and create an issue assigned to M5 so that the removal is not forgotten).